tests/admin/controller/test_metadata_service_self_tests.py

```diff
@@ -0,0 +1,95 @@
+from nose.tools import (
+    set_trace,
+    eq_,
+    assert_raises
+)
+from flask_babel import lazy_gettext as _
+from core.selftest import (
+    HasSelfTests,
+    SelfTestResult,
+)
+from test_controller import SettingsControllerTest
+from core.model import (
+    create,
+    ExternalIntegration,
+)
+from core.opds_import import MetadataWranglerOPDSLookup
+
+from api.admin.problem_details import *
+from api.nyt import NYTBestSellerAPI
+
+class TestMetadataServiceSelfTests(SettingsControllerTest):
+
+    def test_metadata_service_self_tests_with_no_identifier(self):
```

Comment:
I copied these tests from `test_search_service_self_tests.py`. All these tests for the different types of self-test providers take a long time to run, and one great side effect of just testing `SelfTestsController` is almost all of them will go away.